### DISCUSSION TITLE
CTM with random effects and MCMC

### DISCUSSION THREAD

**Discussion by julianlohmann** [cdriveraus/ctsem#26] [Q&A]
Hi Charles,

I am trying to compute a CTM with random effects using MCMC in ctsem (3.5.3). When I set only optimize=FALSE and nopriors=FALSE, I got quite strange estimates, implausible and quite different from the ML estimates. Then I additionally set intoverpop to =TRUE and got quite plausible results. Is it correct that I have to set intoverpop=TRUE for a MCMC model with random effects and what is the purpose of this argument?

Thanks a lot & best regards
Julian

**Comment by cdriveraus** [comment]
Hi Julian. 'integrate over population' uses an extended Kalman filter approach to determine the individual specific parameters, rather than estimating them directly. This is computationally slower, but generally has much simpler posterior geometry and can end up giving more effective samples per second because it is not as tricky for the sampler to explore the space. It *is* based on a linearisation / approximation approach, and will not return exactly the same results as sampling the individual parameters, that is certain. Does your mcmc fit with the strange estimates appear to have converged, ie good Rhat and Neff? If so, then if you're able to post or email an example I can run (maybe with simulated data or whatever works) then I can maybe see what is up.

**Reply by julianlohmann** [reply]
Hi Charles, that's very interesting! Thanks for your explanations. intoverpop=FALSE took significantly longer (certainly five times longer) in my case (data with N=60, T=7) with the same number of iterations and yes, n_eff was considerably smaller compared to intoverpop=TRUE with the same number of iterations and both neff and rhat were not in an exaptable range (but were for intoverpop=TRUE). So, I should try intoverpop=FALSE with much more iterations... The maximum I have tried so far with intoverpop=FALSE is 4000, 3 chains and that took hours, so all in all, I thought something was wrong using intoverpop=FALSE.
If you want I can create and upload a dataset where this occurs.

**Reply by cdriveraus** [reply]
It will be slow yeah, though 4000 iterations has in my experience been enough to get decent results unless there is some problem with the model / data / code . You could try setting the sdscale element of the mymodel$pars object to much smaller values, ie 0.01 or 0.1, at least as a test. This tightens the prior for the standard deviation of the random effects / individual specific parameters. If you have a lot of random effects with only T=7 then I am not surprised it is having trouble, and you will probably need to compromise in some way either through less random effects, huge sampling times / iterations, or tighter priors.
